snuba_sdk/request.py

```diff
@@ -81,9 +81,9 @@ def to_dict(self) -> dict[str, object]:
             serialized_mql = self.query.serialize()
             assert isinstance(serialized_mql, dict)  # mypy
             mql_context = serialized_mql["mql_context"]
-            query = str(serialized_mql["mql"])
+            query: Union[str, Dict[str, Any]] = str(serialized_mql["mql"])
```

Comment:
let's put this under `isinstance(self.query, DeleteQuery)` since the metrics query would actually never be a dict. and then we can add a comment about why the query is different for deletes than the other queries